lib/improver/metadata/utilities.py

```diff
@@ -44,16 +42,14 @@ def generate_hash(data_in):
     Args:
         data_in (any):
             The data from which a hash is to be generated. This can be of any
-            type that can be pickled.
+            type that can be dumped to JSON (dict, list, str, float, etc).
     Returns:
         hash (str):
-            A hexidecimal hash representing the data.
+            A hexadecimal string which is a hash hexdigest of the data as a
+            string.
     """
-    hashable_type = pickle.dumps(data_in)
-    # Marked as 'nosec' as the usage of MD5 hash is to produce a good checksum,
-    # rather than for cryptographic hashing purposes
-    hash_result = hashlib.md5(hashable_type).hexdigest()  # nosec
-    return hash_result
+    bytestring = pprint.pformat(data_in).encode('utf-8')
+    return hashlib.sha256(bytestring).hexdigest()
```

Comment:
(It probably doesn't matter, because it's not hashing a lot of data, and I'd guess isn't being called all that often, but md5 will be quicker than sha256)

Reply:
I agree md5 more suitable - but yes, what the hey, sha256 doesn't flag up anything.